include/nbl/video/IDeviceMemoryBacked.h

```diff
@@ -49,11 +80,29 @@ class IDeviceMemoryBacked : public virtual core::IReferenceCounted
         //! Returns the offset in the allocation at which it is bound to the resource
         virtual size_t getBoundMemoryOffset() const = 0;
 
+        //! For constructor parameter only
+        struct SCreationParams : SCachedCreationParams
+        {
+            const uint32_t* queueFamilyIndices = nullptr;
+        };
+
     protected:
-        IDeviceMemoryBacked() {}
-        IDeviceMemoryBacked(const SDeviceMemoryRequirements& reqs) : cachedMemoryReqs(reqs) {}
+        inline IDeviceMemoryBacked(SCreationParams&& _creationParams, const SDeviceMemoryRequirements& reqs)
+            : m_cachedCreationParams(std::move(_creationParams)), m_cachedMemoryReqs(reqs) {}
+        inline virtual ~IDeviceMemoryBacked()
+        {
+            assert(!m_cachedCreationParams.preDestroyStep); // derived class should have already cleared this out
+        }
+
+        // it's the derived class' responsibility to call this in its destructor
+        inline void preDestroyStep()
```

Comment:
Do we need another function for `postDestroyStep`?

Reply:
we don't because the destructor of `unique_ptr` member takes care of that

Reply:
Oh right